### ISSUE TITLE
PKC looses settings every day

### ISSUE THREAD

**Issue by mvanbaak** [croneter/PlexKodiConnect#499]
Hi man,

I really don't know how I should start debugging this but:
Every day when I boot my raspberrypi with PKC, I have to redo the whole setup.
The first time this happened, was after I upgraded to 2.1.0, redid all the settings and import, and was greeted with the initial setup prompts the next day.
Tested all releases after that, both the stable and beta versions and they all keep doing this.
Also switched between LibreElec and OSMC to rule out the 'distribution'

Yesterday I once again wiped the SD card, installed latest stable OSMC, installed PKC 2.2.2, ran the setup, watched the two latest episodes of a series, went to bed, and this morning I chanded the keymaps/remote.xml to disable the power-off button of my remote. I rebooted the pi to have the changes in effect and there it is again, the initial setup prompts of PKC.

Log of kodi starting up:
```
11:49:27.041 T:1926130176  NOTICE: special://profile/ is mapped to: special://masterprofile/
11:49:27.041 T:1926130176  NOTICE: -----------------------------------------------------------------------
11:49:27.041 T:1926130176  NOTICE: Starting Kodi (17.6). Platform: Linux ARM (Thumb) 32-bit
11:49:27.041 T:1926130176  NOTICE: Using Release Kodi x32 build (version for Raspberry Pi)
11:49:27.041 T:1926130176  NOTICE: Kodi compiled Jun  7 2018 by GCC 6.3.0 for Linux ARM (Thumb) 32-bit version 4.9.30 (264478)
11:49:27.041 T:1926130176  NOTICE: Running on Open Source Media Center 2018.06-1, kernel: Linux ARM 32-bit version 4.14.34-2-osmc
11:49:27.041 T:1926130176  NOTICE: FFmpeg version/source: ffmpeg-3.1-kodi
11:49:27.041 T:1926130176  NOTICE: Host CPU: ARMv7 Processor rev 4 (v7l), 4 cores available
11:49:27.041 T:1926130176  NOTICE: ARM Features: Neon disabled
11:49:27.041 T:1926130176  NOTICE: special://xbmc/ is mapped to: /usr/share/kodi
11:49:27.041 T:1926130176  NOTICE: special://xbmcbin/ is mapped to: /usr/lib/kodi
11:49:27.041 T:1926130176  NOTICE: special://xbmcbinaddons/ is mapped to: /usr/lib/kodi/addons
11:49:27.041 T:1926130176  NOTICE: special://masterprofile/ is mapped to: /home/osmc/.kodi/userdata
11:49:27.041 T:1926130176  NOTICE: special://envhome/ is mapped to: /home/osmc
11:49:27.041 T:1926130176  NOTICE: special://home/ is mapped to: /home/osmc/.kodi
11:49:27.041 T:1926130176  NOTICE: special://temp/ is mapped to: /home/osmc/.kodi/temp
11:49:27.041 T:1926130176  NOTICE: special://logpath/ is mapped to: /home/osmc/.kodi/temp
11:49:27.041 T:1926130176  NOTICE: The executable running is: /usr/lib/kodi/kodi.bin
11:49:27.041 T:1926130176  NOTICE: Local hostname: kodilivingroomtv
11:49:27.041 T:1926130176  NOTICE: Log File is located: /home/osmc/.kodi/temp//kodi.log
11:49:27.041 T:1926130176  NOTICE: -----------------------------------------------------------------------
11:49:27.239 T:1926130176   ERROR: DBus: Error org.freedesktop.DBus.Error.ServiceUnknown - The name org.freedesktop.UPower was not provided by any .service files
11:49:27.430 T:1926130176  NOTICE: load settings...
11:49:27.541 T:1926130176  NOTICE: Found 2 Lists of Devices
11:49:27.541 T:1926130176  NOTICE: Enumerated PI devices:
11:49:27.541 T:1926130176  NOTICE:     Device 1
11:49:27.541 T:1926130176  NOTICE:         m_deviceName      : HDMI
11:49:27.541 T:1926130176  NOTICE:         m_displayName     : HDMI
11:49:27.541 T:1926130176  NOTICE:         m_displayNameExtra:
11:49:27.541 T:1926130176  NOTICE:         m_deviceType      : AE_DEVTYPE_HDMI
11:49:27.542 T:1926130176  NOTICE:         m_channels        : FL,FR
11:49:27.542 T:1926130176  NOTICE:         m_sampleRates     : 8000,11025,16000,22050,24000,32000,44100,48000,88200,96000,176400,192000
11:49:27.542 T:1926130176  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32NE,AE_FMT_S16NE,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP,AE_FMT_RAW
11:49:27.542 T:1926130176  NOTICE:         m_streamTypes     : STREAM_TYPE_AC3,STREAM_TYPE_EAC3,STREAM_TYPE_DTSHD_CORE,STREAM_TYPE_DTS_2048,STREAM_TYPE_DTS_1024,STREAM_TYPE_DTS_512
11:49:27.542 T:1926130176  NOTICE:     Device 2
11:49:27.542 T:1926130176  NOTICE:         m_deviceName      : Analogue
11:49:27.542 T:1926130176  NOTICE:         m_displayName     : Analogue
11:49:27.542 T:1926130176  NOTICE:         m_displayNameExtra:
11:49:27.542 T:1926130176  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
11:49:27.542 T:1926130176  NOTICE:         m_channels        : FL,FR
11:49:27.542 T:1926130176  NOTICE:         m_sampleRates     : 48000
11:49:27.542 T:1926130176  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP
11:49:27.542 T:1926130176  NOTICE:         m_streamTypes     : No passthrough capabilities
11:49:27.542 T:1926130176  NOTICE:     Device 3
11:49:27.542 T:1926130176  NOTICE:         m_deviceName      : Both
11:49:27.542 T:1926130176  NOTICE:         m_displayName     : HDMI and Analogue
11:49:27.542 T:1926130176  NOTICE:         m_displayNameExtra:
11:49:27.542 T:1926130176  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
11:49:27.542 T:1926130176  NOTICE:         m_channels        : FL,FR
11:49:27.542 T:1926130176  NOTICE:         m_sampleRates     : 48000
11:49:27.542 T:1926130176  NOTICE:         m_dataFormats     : AE_FMT_FLOAT,AE_FMT_S32LE,AE_FMT_S16LE,AE_FMT_FLOATP,AE_FMT_S32NEP,AE_FMT_S16NEP
11:49:27.542 T:1926130176  NOTICE:         m_streamTypes     : No passthrough capabilities
11:49:27.542 T:1926130176  NOTICE: Enumerated ALSA devices:
11:49:27.542 T:1926130176  NOTICE:     Device 1
11:49:27.542 T:1926130176  NOTICE:         m_deviceName      : @
11:49:27.542 T:1926130176  NOTICE:         m_displayName     : Default (bcm2835 ALSA bcm2835 ALSA)
11:49:27.542 T:1926130176  NOTICE:         m_displayNameExtra:
11:49:27.542 T:1926130176  NOTICE:         m_deviceType      : AE_DEVTYPE_PCM
11:49:27.542 T:1926130176  NOTICE:         m_channels        : FL,FR
11:49:27.542 T:1926130176  NOTICE:         m_sampleRates     : 8000,11025,16000,22050,32000,44100,48000
11:49:27.542 T:1926130176  NOTICE:         m_dataFormats     : AE_FMT_S16NE,AE_FMT_S16LE,AE_FMT_U8
11:49:27.542 T:1926130176  NOTICE:         m_streamTypes     : No passthrough capabilities
11:49:27.556 T:1926130176  NOTICE: Loaded settings file from special://xbmc/system/advancedsettings.xml
11:49:27.556 T:1926130176  NOTICE: Contents of special://xbmc/system/advancedsettings.xml are...
                                            <?xml version="1.0" encoding="UTF-8" ?>
                                            <!-- Do not modify! These are OSMC Optimised Settings for your device  !-->
                                            <!-- If you wish to override any of these values, then create a file in /home/osmc/.kodi/userdata/advancedsettings.xml !-->
                                            <!-- It will override these settings, and your settings will be preserved across OSMC updates !-->
                                            <advancedsettings>
                                              <imageres>540</imageres>
                                              <fanartres>720</fanartres>
                                              <splash>false</splash>
                                              <handlemounting>0</handlemounting>
                                              <samba>
                                                <clienttimeout>30</clienttimeout>
                                              </samba>
                                            </advancedsettings>
11:49:27.565 T:1926130176  NOTICE: Loaded settings file from special://profile/advancedsettings.xml
11:49:27.565 T:1926130176  NOTICE: Contents of special://profile/advancedsettings.xml are...
                                            <?xml version="1.0" encoding="UTF-8" ?>
                                            <advancedsettings>
                                              <musiclibrary>
                                                <backgroundupdate>true</backgroundupdate>
                                              </musiclibrary>
                                              <videolibrary>
                                                <cleanonupdate>false</cleanonupdate>
                                              </videolibrary>
                                              <video>
                                                <ignorepercentatend>10</ignorepercentatend>
                                                <playcountminimumpercent>90</playcountminimumpercent>
                                                <ignoresecondsatstart>60</ignoresecondsatstart>
                                              </video>
                                              <audio>
                                                <excludefromscan>
                                                  <regexp>^/mnt/mvanbaaknas/Music/</regexp>
                                                </excludefromscan>
                                              </audio>
                                            </advancedsettings>
11:49:27.566 T:1926130176  NOTICE: Default Video Player: VideoPlayer
11:49:27.566 T:1926130176  NOTICE: Default Audio Player: paplayer
11:49:27.566 T:1926130176  NOTICE: Disabled debug logging due to GUI setting. Level 0.
11:49:27.566 T:1926130176  NOTICE: Log level changed to "LOG_LEVEL_NORMAL"
11:49:27.566 T:1926130176  NOTICE: CMediaSourceSettings: loading media sources from special://masterprofile/sources.xml
11:49:27.568 T:1926130176  NOTICE: Loading player core factory settings from special://xbmc/system/playercorefactory.xml.
11:49:27.570 T:1926130176  NOTICE: Loaded playercorefactory configuration
11:49:27.570 T:1926130176  NOTICE: Loading player core factory settings from special://masterprofile/playercorefactory.xml.
11:49:27.570 T:1926130176  NOTICE: special://masterprofile/playercorefactory.xml does not exist. Skipping.
11:49:27.738 T:1926130176  NOTICE: Running database version Addons27
11:49:28.722 T:1926130176  NOTICE: ADDONS: Using repository repository.xbmc.org
11:49:28.722 T:1926130176  NOTICE: ADDONS: Using repository repository.plexkodiconnectbeta
11:49:29.003 T:1926130176  NOTICE: Raspberry PI firmware version: Apr 26 2018 23:18:12 
                                            Copyright (c) 2012 Broadcom
                                            version 90a06d6c6299583cfbf40033bbf895a45920fa71 (clean) (release)
11:49:29.003 T:1926130176  NOTICE: ARM mem: 768MB GPU mem: 256MB MPG2:0 WVC1:0
11:49:29.003 T:1926130176  NOTICE: cache.memorysize: 20MB libass.cache: 0MB
11:49:29.004 T:1926130176  NOTICE: Config:
                                            aphy_params_current=819
                                            arm_freq=1200
                                            audio_pwm_mode=514
                                            config_hdmi_boost=5
                                            core_freq=400
                                            desired_osc_freq=0x36ee80
                                            disable_commandline_tags=2
                                            disable_l2cache=1
                                            disable_splash=1
                                            display_hdmi_rotate=-1
                                            display_lcd_rotate=-1
                                            dphy_params_current=547
                                            force_eeprom_read=1
                                            force_pwm_open=1
                                            framebuffer_ignore_alpha=1
                                            framebuffer_swap=1
                                            gpu_freq=300
                                            hdmi_channel_map=8
                                            hdmi_force_cec_address=65535
                                            hdmi_ignore_cec_init=1
                                            init_uart_clock=0x2dc6c00
                                            lcd_framerate=60
                                            over_voltage_avs=0x27ac4
                                            pause_burst_frames=1
                                            program_serial_random=1
                                            sdram_freq=450
11:49:29.004 T:1926130176  NOTICE: Config:
                                            device_tree=-
11:49:29.018 T:1926130176  NOTICE: InitWindowSystem: Using EGL Implementation: raspberrypi
11:49:29.024 T:1926130176  NOTICE: Found resolution 640 x 480 for display 0 with 640 x 480 @ 60.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 640 x 480 for display 0 with 640 x 480 @ 59.940063 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 60.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 720 x 480 for display 0 with 720 x 480 @ 59.940063 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 60.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 59.940063 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 60.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 59.940063 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 720 x 576 for display 0 with 720 x 576 @ 50.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1280 x 720 for display 0 with 1280 x 720 @ 50.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 50.000000 Hz
11:49:29.024 T:1926130176  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080i @ 60.000000 Hz
11:49:29.025 T:1926130176  NOTICE: Found (1920x1080i@60.000000) at 27, setting to RES_DESKTOP at 16
11:49:29.025 T:1926130176  NOTICE: Checking resolution 22
11:49:29.329 T:1926130176  NOTICE: GL_VENDOR = Broadcom
11:49:29.329 T:1926130176  NOTICE: GL_RENDERER = VideoCore IV HW
11:49:29.329 T:1926130176  NOTICE: GL_VERSION = OpenGL ES 2.0
11:49:29.329 T:1926130176  NOTICE: GL_SHADING_LANGUAGE_VERSION = OpenGL ES GLSL ES 1.00
11:49:29.329 T:1926130176  NOTICE: GL_EXTENSIONS = GL_OES_compressed_ETC1_RGB8_texture GL_OES_compressed_paletted_texture GL_OES_texture_npot GL_OES_depth24 GL_OES_vertex_half_float GL_OES_EGL_image GL_OES_EGL_image_external GL_EXT_discard_framebuffer GL_OES_rgb8_rgba8 GL_OES_depth32 GL_OES_mapbuffer GL_EXT_texture_format_BGRA8888 GL_APPLE_rgb_422 GL_EXT_debug_marker
11:49:30.480 T:1905259264  NOTICE: Running database version Addons27
11:49:30.485 T:1905259264  NOTICE: Running database version ViewModes6
11:49:30.492 T:1905259264  NOTICE: Running database version Textures13
11:49:30.517 T:1905259264  NOTICE: Running database version MyMusic60
11:49:30.544 T:1905259264  NOTICE: Running database version MyVideos107
11:49:30.552 T:1905259264  NOTICE: Running database version TV29
11:49:30.560 T:1905259264  NOTICE: Running database version Epg11
11:49:30.562 T:1798304512  NOTICE: Register - new cec device registered on cec->RPI: CEC Adapter (2708:1001)
11:49:30.567 T:1926130176  NOTICE: start dvd mediatype detection
11:49:31.609 T:1926130176 WARNING: JSONRPC: Could not parse type "Setting.Details.SettingList"
11:49:31.692 T:1926130176 WARNING: CApplication::Initialize - startup.xml taints init process
11:49:31.737 T:1926130176  NOTICE: initialize done
11:49:31.737 T:1926130176  NOTICE: Running the application...
11:49:31.743 T:1926130176  NOTICE: starting zeroconf publishing
11:49:31.744 T:1926130176  NOTICE: CWebServer[80]: Started
11:49:31.744 T:1926130176  NOTICE: starting upnp client
11:49:31.777 T:1613755136  NOTICE: ES: Starting UDP Event server on port 9777
11:49:31.777 T:1613755136  NOTICE: UDP: Listening on port 9777 (ipv6 : true)
11:49:33.591 T:1926130176   ERROR: ExecuteAsync - Not executing non-existing script script.favourites
11:49:45.675 T:1706029824  NOTICE: PLEX.service_entry: Delaying Plex startup by: 0 sec...
11:49:45.758 T:1706029824  NOTICE: PLEX.service_entry: ======== START PlexKodiConnect ========
11:49:45.759 T:1706029824  NOTICE: PLEX.service_entry: Platform: Linux
11:49:45.760 T:1706029824  NOTICE: PLEX.service_entry: KODI Version: 17.6
11:49:45.761 T:1706029824  NOTICE: PLEX.service_entry: PlexKodiConnect Version: 2.2.2
11:49:45.776 T:1706029824  NOTICE: PLEX.service_entry: PKC Direct Paths: False
11:49:45.789 T:1706029824  NOTICE: PLEX.service_entry: Number of sync threads: 10
11:49:45.791 T:1706029824  NOTICE: PLEX.service_entry: Playlist m3u encoding: utf-8
11:49:45.793 T:1706029824  NOTICE: PLEX.service_entry: Full sys.argv received: ['']
11:49:45.794 T:1706029824  NOTICE: PLEX.initialsetup: Start (re-)loading PKC settings
11:49:47.281 T:1706029824  NOTICE: PLEX.videonodes: Clearing nodes properties.
11:49:47.608 T:1706029824  NOTICE: PLEX.clientinfo: Generating a new deviceid.
11:49:49.981 T:1706029824  NOTICE: PLEX.clientinfo: Unique device Id plex_client_Id generated: f2b341db-f473-4c13-bf31-7e07bd302c36
11:49:50.334 T:1706029824  NOTICE: PLEX.initialsetup: Done (re-)loading PKC settings
11:49:50.654 T:1706029824  NOTICE: PLEX.initialsetup: Initial setup called.
11:49:50.658 T:1697641216  NOTICE: Skin Helper Service --> Skin Helper Service version 1.1.3 started
11:49:50.658 T:1431409408  NOTICE: Skin Helper Service --> WebService - start helper webservice on port 52307
11:49:50.741 T:1706029824  NOTICE: PLEX.initialsetup: Current Kodi video memory cache in bytes: 20971520
11:49:51.364 T:1706029824  NOTICE: PLEX.migration: Checking whether we need to migrate something
11:49:51.376 T:1706029824  NOTICE: PLEX.utils: current DB:  minimum DB: 1.8.2
11:49:51.378 T:1706029824  NOTICE: PLEX.utils: current DB:  minimum DB: 2.0.25
11:49:55.724 T:1490023168  NOTICE: PLEX.default: plugin.video.plexkodiconnect started
11:49:57.426 T:1706029824  NOTICE: PLEX.plex_tx: Successfully retrieved code and id from plex.tv
11:49:58.354 T:1490023168  NOTICE: PLEX.default: plugin.video.plexkodiconnect stopped
11:50:01.003 T:1529869056  NOTICE: PLEX.default: plugin.video.plexkodiconnect started
11:50:05.496 T:1714418432 WARNING: CSkinInfo: failed to load skin settings
11:50:07.586 T:1529869056  NOTICE: PLEX.default: plugin.video.plexkodiconnect stopped
```

**Comment by croneter** [comment]
That's extremely weird... I myself am using LibreElec with a Raspberry Pi 3 - never had these issues. 

I don't see anything in the log excerpt. Could you please enable debug logging and share the entire log kodi.log plus kodi.log.old (or something like it) once this happens again?

You DID try another SD card, right?

**Comment by mvanbaak** [comment]
Yes, tried 3 SD cards.
I now have this issue on every reboot.
Will enable debug logging and add them

**Comment by croneter** [comment]
All I'm seeing is this:
```
WARNING: CSkinInfo: failed to load skin settings
```
which lead to this: https://discourse.osmc.tv/t/skin-changes-lost-after-a-reboot/21834/10 The thread is mainly talking about skins, but a skin is also just an add-on and skin settings work the same as other add-on settings. 

Definitely NOT related to PlexKodiConnect, sorry to tell you. Please take it up with OSMC. Thanks if you could report back here with the solution that worked...

**Comment by mvanbaak** [comment]
Well, as stated in my original bug report, I started to see this behaviour on LibreElec and switched to OSMC to see if it was something LibreElec specific.

I am once again switching to a clean and new install of LibreElec and I'll setup the stable PKC on it to see if it happens again.

**Comment by croneter** [comment]
Thing is: if PKC's settings.xml is missing, Kodi will automatically recreate it. Where is your [Kodi userdata](https://kodi.wiki/view/Userdata) located, exactly, what's the path?

**Comment by mvanbaak** [comment]
it's in /home/osmc/.kodi/userdata/addon_data/plugin.video.plexkodiconnect/settings.xml

**Comment by croneter** [comment]
OK, thanks, so no funny non-ASCII characters there either. 

PKC does not "directly" mess with settings.xml (does not open the file directly), but uses Kodi python commands to do that. If the settings get screwed up, I don't see how that can be PKC's doing... 

I'm rebooting my Raspberry Pi here with the latest beta, using LibreElec. Settings stick.

**Comment by mvanbaak** [comment]
I'm giving up. Either my raspberry or my sd cards are toast. Must be it.

Next week I'll pickup an old laptop at my dad's place and use that one ;-)

**Comment by mvanbaak** [comment]
Hum, can it be that the Music library is causing this?
I am testing with a fresh install and I already got 2 times a popup stating a new Music library was found, kodi had to restart and it started the initial setup again.

Logs:
```
17:43:21.017 T:1367929760  NOTICE: PLEX.music: New Plex music library detected: ^/mnt/mvanbaaknas/Music/
17:43:21.163 T:1367929760  NOTICE: PLEX.music: Deleting music library from advancedsettings: ^/storage/mvanbaaknas/Music/
17:43:27.168 T:1944929952 WARNING: Label Formating: $VAR[WeatherFanArtSingleImage_Current] is not defined
17:43:27.272 T:1599271840   ERROR: GetDirectory - Error getting library://video/Plex-2/2_ondeck.xml/
17:43:27.566 T:1857024928   ERROR: GetDirectory - Error getting library://video/Plex-1/1_ondeck.xml/
17:43:27.568 T:1599271840   ERROR: GetDirectory - Error getting library://video/Plex-5/5_ondeck.xml/
17:43:27.569 T:1857024928   ERROR: GetDirectory - Error getting library://video/Plex-6/6_ondeck.xml/
17:43:27.569 T:1857024928   ERROR: GetDirectory - Error getting noop
17:43:27.691 T:1944929952  NOTICE: Storing total System Uptime
17:43:27.691 T:1944929952  NOTICE: Saving settings
17:43:27.706 T:1944929952  NOTICE: Saving skin settings
17:43:27.713 T:1944929952  NOTICE: stop all
17:43:27.729 T:1367929760  NOTICE: PLEX.videonodes: Clearing nodes properties.
17:43:27.731 T:1750266784  NOTICE: PLEX.kodimonitor: Kodi OnQuit detected - shutting down
17:43:27.760 T:1750266784  NOTICE: PLEX.service: ======== STOP PlexKodiConnect ========
17:43:27.937 T:1332949920  NOTICE: PLEX.playqueue: ----===## PlayqueueMonitor stopped ##===----
17:43:27.938 T:1394520992  NOTICE: PLEX.userclient: ##===---- UserClient Stopped ----===##
17:43:27.938 T:1359541152  NOTICE: PLEX.plexbmchelper.subscribers: Signaling a complete stop to PMS
17:43:27.941 T:1528820640  NOTICE: PLEX.command_pipeline: ----===## Kodi_Play_Client stopped ##===----
17:43:27.947 T:1411298208  NOTICE: PLEX.kodimonitor: #====---- Special Monitor Stopped ----====#
17:43:27.958 T:1341338528  NOTICE: PLEX.playback_starter: ----===## PlaybackStarter stopped ##===----
17:43:27.968 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 5 viewname: Adult Movies viewtype: movie tagid: 1
17:43:27.969 T:1367929760  NOTICE: PLEX.utils: Writing playlist file to: /storage/.kodi/userdata/playlists/video/Plex 5.xsp
17:43:27.976 T:1367929760  NOTICE: PLEX.utils: Successfully added playlist: Adult Movies
17:43:28.062 T:1386132384  NOTICE: PLEX.artwork: ---===### Stopped Image_Cache_Thread ###===---
17:43:28.064 T:1376318368  NOTICE: PLEX.websocket_client: ##===---- PMS_Websocket Stopped ----===##
17:43:28.068 T:1421112224  NOTICE: PLEX.websocket_client: ##===---- Alexa_Websocket Stopped ----===##
17:43:28.308 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 6 viewname: Adult TV Shows viewtype: show tagid: 2
17:43:28.309 T:1367929760  NOTICE: PLEX.utils: Writing playlist file to: /storage/.kodi/userdata/playlists/video/Plex 6.xsp
17:43:28.310 T:1367929760  NOTICE: PLEX.utils: Successfully added playlist: Adult TV Shows
17:43:28.408 T:1548940192  NOTICE: PLEX.default: plugin.video.plexkodiconnect stopped
17:43:28.443 T:1944929952  NOTICE: stop player
17:43:28.444 T:1944929952  NOTICE: ES: Stopping event server
17:43:28.606 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 1 viewname: Movies viewtype: movie tagid: 3
17:43:28.607 T:1367929760  NOTICE: PLEX.utils: Writing playlist file to: /storage/.kodi/userdata/playlists/video/Plex 1.xsp
17:43:28.608 T:1367929760  NOTICE: PLEX.utils: Successfully added playlist: Movies
17:43:28.714 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 3 viewname: Music viewtype: artist tagid: 4
17:43:28.714 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 4 viewname: Photos viewtype: photo tagid: 5
17:43:28.717 T:1367929760  NOTICE: PLEX.librarysync: Found viewid: 2 viewname: TV Shows viewtype: show tagid: 6
17:43:28.718 T:1367929760  NOTICE: PLEX.utils: Writing playlist file to: /storage/.kodi/userdata/playlists/video/Plex 2.xsp
17:43:28.719 T:1367929760  NOTICE: PLEX.utils: Successfully added playlist: TV Shows
17:43:28.744 T:1367929760  NOTICE: PLEX.librarysync: Finished processing views. Views saved: [{'itemtype': u'movie', 'kodi_tagid': 1, 'id': u'5', 'sync_to_kodi': 1, 'name': u'Adult Movies'}, {'itemtype': u'show', 'kodi_tagid': 2, 'id': u'6', 'sync_to_kodi': 1, 'name': u'Adult TV Shows'}, {'itemtype': u'movie', 'kodi_
tagid': 3, 'id': u'1', 'sync_to_kodi': 1, 'name': u'Movies'}, {'itemtype': u'artist', 'kodi_tagid': 4, 'id': u'3', 'sync_to_kodi': 1, 'name': u'Music'}, {'itemtype': u'photo', 'kodi_tagid': 5, 'id': u'4', 'sync_to_kodi': 1, 'name': u'Photos'}, {'itemtype': u'show', 'kodi_tagid': 6, 'id': u'2', 'sync_to_kodi': 1, 'nam
e': u'TV Shows'}]
17:43:28.745 T:1616049056  NOTICE: ES: UDP Event server stopped
17:43:28.745 T:1367929760  NOTICE: PLEX.librarysync: Running fullsync for NEW PMS items with repair=False
17:43:28.745 T:1367929760  NOTICE: PLEX.utils: It took 0:00:00.000539 to run the function full_sync
17:43:28.746 T:1367929760  NOTICE: PLEX.librarysync: Startup sync has not yet been successful
17:43:28.776 T:1359541152  NOTICE: PLEX.plexbmchelper.plexgdm: Discovery shutting down
17:43:28.857 T:1367929760  NOTICE: PLEX.downloadutils: Request session stopped
17:43:28.858 T:1367929760  NOTICE: PLEX.librarysync: ###===--- LibrarySync Stopped ---===###
17:43:28.867 T:1944929952  NOTICE: stopping upnp
17:43:29.018 T:1944929952  NOTICE: stopping zeroconf publishing
17:43:29.024 T:1944929952  NOTICE: CWebServer[8080]: Stopped
17:43:29.024 T:1944929952  NOTICE: stop dvd detect media
17:43:29.091 T:1359541152  NOTICE: PLEX.plexbmchelper.plexgdm: Registration shutting down
17:43:29.255 T:1402909600  NOTICE: PLEX.plexbmchelper.plexgdm: Client Update loop stopped
17:43:29.257 T:1359541152  NOTICE: PLEX.PlexCompanion: ----===## Plex Companion stopped ##===----
17:43:29.508 T:1750266784 WARNING: CPythonInvoker(0, /storage/.kodi/addons/plugin.video.plexkodiconnect/service.py): the python script "/storage/.kodi/addons/plugin.video.plexkodiconnect/service.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon9xbmcaddon5AddonE
17:43:31.604 T:1944929952  NOTICE: clean cached files!
17:43:31.749 T:1741878176  NOTICE: Skin Helper Service --> Shutdown requested !
17:43:31.756 T:1741878176  NOTICE: Skin Helper Service --> Skin Helper Service version 1.1.3 stopped
17:43:35.186 T:1733489568  NOTICE: Skin Helper Backgrounds --> BackgroundsUpdater - stop called
17:43:35.311 T:1725100960  NOTICE: Skin Helper Widgets --> Backgroundservice stopped
17:43:35.432 T:1944929952  NOTICE: closing down remote control service
17:43:35.432 T:1944929952  NOTICE: stopped
17:43:35.633 T:1944929952  NOTICE: destroy
17:43:35.633 T:1944929952  NOTICE: unload skin
17:43:35.670 T:1944929952 WARNING: Cleanup: Having to cleanup texture common/white.png
17:43:35.670 T:1944929952 WARNING: Cleanup: Having to cleanup texture /usr/share/kodi/addons/service.libreelec.settings/resources/skins/Default/media/default/button-fo.png
17:43:35.670 T:1944929952 WARNING: Cleanup: Having to cleanup texture /usr/share/kodi/addons/service.libreelec.settings/resources/skins/Default/media/default/button-alt-nofo.png
17:43:35.671 T:1944929952 WARNING: Cleanup: Having to cleanup texture /usr/share/kodi/addons/service.libreelec.settings/resources/skins/Default/media/default/Logo.png
17:43:35.745 T:1944929952  NOTICE: unload sections
17:43:35.759 T:1944929952  NOTICE: special://profile/ is mapped to: special://masterprofile/
17:43:35.764 T:1944929952  NOTICE: application stopped...
```

**Comment by mvanbaak** [comment]
And the logs after the reboot:
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/2130946/kodi.log)

**Comment by mvanbaak** [comment]
I disabled syncing the Music library. Let's see how it works out this way

**Comment by croneter** [comment]
Could be anything, really :-(

**Comment by mvanbaak** [comment]
```
17:43:21.017 T:1367929760  NOTICE: PLEX.music: New Plex music library detected: ^/mnt/mvanbaaknas/Music/
17:43:21.163 T:1367929760  NOTICE: PLEX.music: Deleting music library from advancedsettings: ^/storage/mvanbaaknas/Music/
```

This was in the log both times it restarted and after the restart wanted to do the initial setup thing again.
That's why I think it's the music library

**Comment by Pavuucek** [comment]
I've been getting this issue lately and I've found something strange in pkc settings.xml.
I have these two lines:
`
<setting id="caching_artwork_count" value="$ADDON[plugin.video.plexkodiconnect 39310]" />
<setting id="fanarttv_lookups" value="$ADDON[plugin.video.plexkodiconnect 39310]" />
`
... and I don't think kodi actually undrestands those values, because changing them to something meaningful, like "0" makes kodi actually load and save settings properly. Leaving them causes settings reset every reboot.
@mvanbaak Could you please check your settings file has these lines and what does the value part actually contain?

**Comment by croneter** [comment]
I'm just wondering why this did not happen to me...

**Comment by mvanbaak** [comment]
@croneter are you using path substitution?
If I enable the music library sync, it adds the substituded path to advancedsettings.xml
If I disable the music library sync, it adds the pkc provided path to the advancedsettings.xml (of course this time in the exclude tag)

I think it has something to do with that. Not sure though.

**Comment by croneter** [comment]
@Pavuucek 
Nope, these settings work fine, even with non-English languages and non-ASCII characters
```
<setting id="caching_artwork_count" value="$ADDON[plugin.video.plexkodiconnect 39310]" />
```
No issue whatsoever if I add these lines to the PKC Kodi settings

**Comment by croneter** [comment]
@mvanbaak 
No matter what I do, I cannot reproduce your issue. Can you figure out when the problem starts? Does it start as soon as you turn on the music library? Does it matter whether you use Direct Paths or Add-on Paths (remember to resync after switching from one to the other)?

**Comment by croneter** [comment]
> ```17:43:21.017 T:1367929760  NOTICE: PLEX.music: New Plex music library detected: ^/mnt/mvanbaaknas/Music/
17:43:21.163 T:1367929760  NOTICE: PLEX.music: Deleting music library from advancedsettings: ^/storage/mvanbaaknas/Music/```
>This was in the log both times it restarted and after the restart wanted to do the initial setup thing again.
That's why I think it's the music library

> As you might guess from that notice: I'm using direct paths and rewrite the paths.
All media types have this as path override:
/mnt/ -> /storage/

>My plex server has all the data in /mnt/mvanbaaknas/ and that same share is mounted under /storage/mvanbaaknas/ on my raspberry

**Comment by Pavuucek** [comment]
@croneter Yes, I tried to reproduce this thing with clean profile and so far nothing... (YET :-) ) So it must be skin or some other addon causing a mess. Could you please at least limit writing stuff to settings to only what's really necessary? Just in case.
Also, these two setting's default values don't work properly. This is what pkc settings dialog looked like right after installing.
![pkc](https://user-images.githubusercontent.com/1253845/41931078-3e832632-797d-11e8-9e04-567f37353aa7.PNG)

**Comment by croneter** [comment]
@Pavuucek Will do, thanks for the info. Settings are already kept at a minimum, but they're not causing this.

I need to be able to reproduce the bug in order to solve, unfortunately

**Comment by Pavuucek** [comment]
So, after a few days of nothing, settings losses started happening again to me. So I modified settings function in utils.py to log what's actually going on:

`def settings(setting, value=None):
    """
    Get or add addon setting. Returns unicode

    setting and value can either be unicode or string
    """
    # We need to instantiate every single time to read changed variables!
    addon = xbmcaddon.Addon(id='plugin.video.plexkodiconnect')
    if value is not None:
        # Takes string or unicode by default!
        addon.setSetting(try_encode(setting), try_encode(value))
        LOG.info('pkc setting: %s to %s', setting, value)
    else:
        # Should return unicode by default, but just in case
        LOG.info('pkc getting: %s = %s', setting, try_decode(addon.getSetting(setting)))
        return try_decode(addon.getSetting(setting))`

And I managed to record a settings loss happen, but sadly I didn't have debugging on.  At least it's something.  I'll send debug logs when it happens again...
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/2143011/kodi.log)

**Comment by croneter** [comment]
Please try the new beta version 2.2.6. Let me know if that does not help and reopen the issue here. Thx!

**Comment by croneter** [comment]
@Pavuucek 
The log was not really showing anything, by the way... :-/ But thanks anyway!

**Comment by Pavuucek** [comment]
So, it happened againand this time I had debugging on, so here's a big logfile.
Settings loss happened around 19:35:38.096. But I'm afraid it still doesn't show about why it happened.
[kodi.zip](https://github.com/croneter/PlexKodiConnect/files/2146586/kodi.zip)

**Comment by croneter** [comment]
Darn! Do you know when this issue started happening? I'm really at a loss here

**Comment by croneter** [comment]
How do you loose settings? Only on reboot? Or just like that?

**Comment by Pavuucek** [comment]
I initially thought it happens on reboot, but that's just when pkc notices it's missing login info. Last time it happened I was just watching tv. Strange is, that it happens almost every day, but only once. I don't know what versuon it started happening, but it was definitely after 2.0.

**Comment by croneter** [comment]
Does disabling Plex Music make a difference?

**Comment by danjames92** [comment]
Pretty sure this started with 2.2.6 for me. Seems random and sporadic. :(

**Comment by Pavuucek** [comment]
So, I spent better part of yesterday to figure what's causing these settings losses and trying If I can reproduce them on my other kodi boxes. No luck other 2 boxes work just fine, nothing suspicious in log files. My primary box (which suffers from settings losses) however started to act more and more weird.

First there were those errors about accessing UPNP devices.
```
19:27:13.124 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "3cf75fa8-82e7-4131-b7e4-28917331aba6" detected @ http://192.168.0.9:9119/screen_sharing
19:27:13.126 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2525-2828-2500-202bc19b1050" detected @ http://169.254.100.156:37215/upnpdev.xml
19:27:13.126 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2626-2828-2600-202bc19b1050" detected @ http://169.254.100.156:37215/tr064dev.xml
19:27:13.627 T:545389900000  NOTICE: Platinum [platinum.core.devicedata]: root namespace is invalid: urn:samsung.com:device-1-0
19:27:13.628 T:545389900000   ERROR: Platinum [platinum.core.devicedata]: NPT_CHECK failed, result=-1 (FAILURE) [((-1))]
19:27:13.628 T:545389900000  NOTICE: Platinum [platinum.core.ctrlpoint]: Device "607e53ba-8cce-a1d7-1ce9-9a5985c6f7b5" is now known as "" (http://192.168.0.9:9119/screen_sharing)
19:28:05.473 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "3cf75fa8-82e7-4131-b7e4-28917331aba6" detected @ http://192.168.0.9:9119/screen_sharing
19:28:05.978 T:545905438944  NOTICE: Platinum [platinum.core.devicedata]: root namespace is invalid: urn:samsung.com:device-1-0
19:28:07.937 T:545905438944   ERROR: Platinum [platinum.core.devicedata]: NPT_CHECK failed, result=-1 (FAILURE) [((-1))]
19:28:07.938 T:545905438944  NOTICE: Platinum [platinum.core.ctrlpoint]: Device "6864d21d-8249-d1a0-6c8e-e68d40136f2c" is now known as "" (http://192.168.0.9:9119/screen_sharing)
19:28:13.678 T:545575182560 WARNING: Platinum [neptune.http]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(m_Connector->Connect(request.GetUrl(), *this, use_proxy?&proxy:__null, http_1_1, connection))]
19:28:13.894 T:545575182560   FATAL: Platinum [platinum.core.ctrlpoint]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(res)]
19:28:13.894 T:545575182560   ERROR: Platinum [platinum.core.ctrlpoint]: Bad Description response @ http://169.254.100.156:37215/upnpdev.xml:
19:28:13.895 T:545356345568 WARNING: Platinum [neptune.http]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(m_Connector->Connect(request.GetUrl(), *this, use_proxy?&proxy:__null, http_1_1, connection))]
19:28:13.895 T:545356345568   FATAL: Platinum [platinum.core.ctrlpoint]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(res)]
19:28:13.895 T:545356345568   ERROR: Platinum [platinum.core.ctrlpoint]: Bad Description response @ http://169.254.100.156:37215/tr064dev.xml:
19:28:57.990 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2525-2828-2500-202bc19b1050" detected @ http://169.254.100.156:37215/upnpdev.xml
19:28:57.991 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2626-2828-2600-202bc19b1050" detected @ http://169.254.100.156:37215/tr064dev.xml
19:28:57.992 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "3cf75fa8-82e7-4131-b7e4-28917331aba6" detected @ http://192.168.0.9:9119/screen_sharing
19:28:58.496 T:545880273120  NOTICE: Platinum [platinum.core.devicedata]: root namespace is invalid: urn:samsung.com:device-1-0
19:28:58.496 T:545880273120   ERROR: Platinum [platinum.core.devicedata]: NPT_CHECK failed, result=-1 (FAILURE) [((-1))]
19:28:58.497 T:545880273120  NOTICE: Platinum [platinum.core.ctrlpoint]: Device "27619b76-eb54-b4ee-02ff-9e35b8a1a89d" is now known as "" (http://192.168.0.9:9119/screen_sharing)
19:29:49.914 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "3cf75fa8-82e7-4131-b7e4-28917331aba6" detected @ http://192.168.0.9:9119/screen_sharing
19:29:50.419 T:545905438944  NOTICE: Platinum [platinum.core.devicedata]: root namespace is invalid: urn:samsung.com:device-1-0
19:29:50.420 T:545905438944   ERROR: Platinum [platinum.core.devicedata]: NPT_CHECK failed, result=-1 (FAILURE) [((-1))]
19:29:50.420 T:545905438944  NOTICE: Platinum [platinum.core.ctrlpoint]: Device "fb2ee203-1647-119e-6d61-f581f8bd87f7" is now known as "" (http://192.168.0.9:9119/screen_sharing)
19:29:58.548 T:545356345568 WARNING: Platinum [neptune.http]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(m_Connector->Connect(request.GetUrl(), *this, use_proxy?&proxy:__null, http_1_1, connection))]
19:29:58.549 T:545356345568   FATAL: Platinum [platinum.core.ctrlpoint]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(res)]
19:29:58.549 T:545356345568   ERROR: Platinum [platinum.core.ctrlpoint]: Bad Description response @ http://169.254.100.156:37215/upnpdev.xml:
19:29:58.549 T:545575182560 WARNING: Platinum [neptune.http]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(m_Connector->Connect(request.GetUrl(), *this, use_proxy?&proxy:__null, http_1_1, connection))]
19:29:58.549 T:545575182560   FATAL: Platinum [platinum.core.ctrlpoint]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(res)]
19:29:58.549 T:545575182560   ERROR: Platinum [platinum.core.ctrlpoint]: Bad Description response @ http://169.254.100.156:37215/tr064dev.xml:
19:30:42.440 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2525-2828-2500-202bc19b1050" detected @ http://169.254.100.156:37215/upnpdev.xml
19:30:42.441 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "00e0fc37-2626-2828-2600-202bc19b1050" detected @ http://169.254.100.156:37215/tr064dev.xml
19:30:42.442 T:545930604768  NOTICE: Platinum [platinum.core.ctrlpoint]: Inspecting device "3cf75fa8-82e7-4131-b7e4-28917331aba6" detected @ http://192.168.0.9:9119/screen_sharing
19:30:42.947 T:545575182560  NOTICE: Platinum [platinum.core.devicedata]: root namespace is invalid: urn:samsung.com:device-1-0
19:30:42.947 T:545575182560   ERROR: Platinum [platinum.core.devicedata]: NPT_CHECK failed, result=-1 (FAILURE) [((-1))]
19:30:42.947 T:545575182560  NOTICE: Platinum [platinum.core.ctrlpoint]: Device "392027e8-015e-9912-6561-a17cd4f274ba" is now known as "" (http://192.168.0.9:9119/screen_sharing)
```
I don't have UPNP enabled on any of my kodi devices and those probably aren't related to pkc in any way.

Then I started getting these sessionKey errors.
```
19:31:27.863 T:545558155488   ERROR: PLEX.librarysync: Processing of Plex Companion message has crashed
19:31:27.863 T:545558155488   ERROR: PLEX.librarysync: Message was: {u'PlaySessionStateNotification': [{u'viewOffset': 0, u'url': u'', u'ratingKey': u'', u'state': u'stopped', u'key': u'', u'guid': u''}], u'type': u'playing', u'size': 1}
19:31:27.865 T:545558155488   ERROR: PLEX.librarysync: Traceback:
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1029, in process_message
                                                self.process_playing(message['PlaySessionStateNotification'])
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1279, in process_playing
                                                session_key = item['sessionKey']
                                            KeyError: 'sessionKey'
19:31:27.977 T:545558155488   ERROR: PLEX.librarysync: Processing of Plex Companion message has crashed
19:31:27.977 T:545558155488   ERROR: PLEX.librarysync: Message was: {u'PlaySessionStateNotification': [{u'viewOffset': 0, u'url': u'', u'ratingKey': u'', u'state': u'stopped', u'key': u'', u'guid': u''}], u'type': u'playing', u'size': 1}
19:31:27.978 T:545558155488   ERROR: PLEX.librarysync: Traceback:
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1029, in process_message
                                                self.process_playing(message['PlaySessionStateNotification'])
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1279, in process_playing
                                                session_key = item['sessionKey']
                                            KeyError: 'sessionKey'
19:31:27.992 T:545558155488   ERROR: PLEX.librarysync: Processing of Plex Companion message has crashed
19:31:27.993 T:545558155488   ERROR: PLEX.librarysync: Message was: {u'PlaySessionStateNotification': [{u'viewOffset': 0, u'url': u'', u'ratingKey': u'', u'state': u'stopped', u'key': u'', u'guid': u''}], u'type': u'playing', u'size': 1}
19:31:27.993 T:545558155488   ERROR: PLEX.librarysync: Traceback:
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1029, in process_message
                                                self.process_playing(message['PlaySessionStateNotification'])
                                              File "/storage/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/librarysync.py", line 1279, in process_playing
                                                session_key = item['sessionKey']
                                            KeyError: 'sessionKey'
```

These probably also don't mean anything, because shortly after I started getting those my whole kodi profile sort of collapsed on itself. Couldn't play ANYTHING, be it PKC content, TV or stuff that cames from other plugins. 
So I ended with reformating my sd card, reinstalling libreelec and restoring profile from other machine. Everything seems to be working properly for now. So there's that :-)

**Comment by croneter** [comment]
@Pavuucek 
Thanks for taking so much time to help troubleshoot. Log messages from other add-ons like this:
```
Platinum [platinum.core.ctrlpoint]: NPT_CHECK failed, result=-20014 (NPT_ERROR_TIMEOUT) [(res)]
```
DO point to some kind of sd card/hdd issue. The PKC KeyError you found is completely unrelated. 

Is anyone here suffering of this issue and does NOT use an SD card but a hard disk?

**Comment by croneter** [comment]
I still cannot reproduce 😞. 

Could anyone who loses settings regurarly on reboot grab `addon_data/plugin.video.plexkodiconnect/settings.xml` from the [Kodi userdata folder](https://kodi.wiki/view/Userdata) just before a reboot while Kodi is still running, then when Kodi is shut down and finally when Kodi did reboot PKC lost all settings? Unfortunately, this will probably ONLY be useful if you loose your settings indeed on the restart. 
* Be sure to delete the values of your token saved under `accessToken` and `plexToken`. 
* You might also want to delete the values of `ipaddress`, `plex_client_Id`


I'd be interested in seeing that file before it somehow gets corrupted.

**Comment by danjames92** [comment]
Just to provide further info... I am using a eMMC 8GB flash card with a advancedsettings.xml tweak to store the video cache in ram so that I don't destroy my flash memory with writes.

**Comment by croneter** [comment]
Hope that fixes it in newest beta version 2.2.7. Please let me know...

**Comment by Pavuucek** [comment]
I finally managed to reproduce this on another device (rpi3) and I came to conslusion that this whole issue was just pkc changing and accessing settings way too often.  My current guess is when InitialSetup gets called multiple times before first call updates settings in kodi's memory the second one resets them to default. Probably :-). It also apparently specific to arm platform because I haven't observed this behavior on a PC.
Anyway, 2.2.7 actually sort-of fixed this. Settings are now saved properly and at least for now, they hold between reboots. So thank you very much :-)
Unfortunately I'm still getting those sessionKey errors I mentioned earlier and they started to appear on all of my devices, so I'll open a new issue.

**Comment by croneter** [comment]
Agreed, @Pavuucek 
My guess is **writing** settings to file "too often" caused this issue. Rather embarassing for Kodi and almost impossible to try to circumvent from a developer perspective 😞 .

**Comment by vinanrra** [comment]
I having same error, but every time i close and open PKC

2020-02-26 16:56:31.506 T:2488877264   ERROR: CAddon[script.plex]: failed to load addon settings from special://profile/addon_data/script.plex/settings.xml
